server/main.go

```diff
@@ -38,6 +38,10 @@ func getWebHandler(activeFileManager *ActiveFileManager, fileStore FileStore) ht
 		method := req.Method
 		path := urlPathToArray(req.URL.Path)
 
+		res.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
> So I realize you're trying to XHR this, but why do we need cross-domain XHRs?

The motivation is described in the PR overview:

> This way, frontend clients are treated the same way as backend clients in their ability to operate.

Basically, right now all backend clients can operate without restrictions. I apply the same behavior for frontend clients. In future, we might want to add client restrictions and those should apply both for backend and frontend clients. Until then, treat them all the same.

> Could we have the InstantShare server serve the instant upload page?

We could, this change enables that. But it's a completely separate task. I don't think it needs to be done now.